### PR TITLE
Venn, Sequence Sun Burst & Geo Spatial diagrams

### PR DESCRIPTION
**Pull request by maqboolkhan**
- [x] Tested manually 
- [x] All 3 diagrams are working 

This branch was also presented to our Historian group

### CODE REVIEW

**Comment by maqboolkhan** [comment]
this PR is ready I double check it, it can be merged it right away

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+            dataMap.put("ssbDiagramData", response);
+            responseBean.setPayload(dataMap);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+		return "pass";
+	}
+
+    private void extractRangData(File inputFile, int col1, int col2) throws IOException {
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: This should probably be renamed.

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        // create csvreaderbuilder for dnstTbl
+        CSVReaderBuilder rBuilder = new CSVReaderBuilder(new FileReader(inputFile));
+        // build the reader
+        CSVReader csvReader = rBuilder.build();
+        try {
+            // read first line
+            csvReader.readNext();
+            int fhrIdIndx = col1;
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            int rangIndx = col2;
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            for (String[] line; (line = csvReader.readNext()) != null;) {
+                int fhrId = Integer.parseInt(line[fhrIdIndx]);
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                int rang = Integer.parseInt(line[rangIndx]);
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                TreeSet<Integer> rangSet = rangMap.get(fhrId);
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                rangSet.add(rang);
+            }
+        } finally {
+            csvReader.close();
+        }
+
+    }
+
+    private void generateSeqCountMap() {
+        for (Integer fhrId : rangMap.keySet()) {
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename fhrId

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+ * SSBDiagramHandler is a subroutine that is used to generate data for
+ * Sequence Sun Burst diagram.
+ *
+ * @author Maqbool
+ */
+
+@Component
+public class SSBDiagramHandler implements Subroutine {
+	@Autowired
+	private FileUtil DemoMain;
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

**Comment by maqboolkhan** [comment]
Thanks, @Cortys but I guess every visualization is data dependent somehow. For example, let's say a simple diagram like Venn diagram will not work also if provided data has nothing common. 
I believe @nikit91 can comment better.

Thanks cc @nikit91

**Comment by nikit91** [comment]
Hi @maqboolkhan ,
Visualizations should be data agnostic and should only have dependency on type of values they could accept.

At the time of development the visualizations were mostly made to target a particular use-case. However, most of the visualizations now can also handle different use cases as well.

In my opinion, we should refactor our variables and methods (if any) to carry generic names.

**Comment by maqboolkhan** [comment]
Thanks, I am done with refactoring with the names

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 9]
```diff
@@ -72,23 +71,23 @@ public String call (com.rivescript.RiveScript rs, String[] args) {
...
-            int fhrIdIndx = col1;
-            int rangIndx = col2;
+            int col1Index = col1;
+            int rangeIndex = col2;
             for (String[] line; (line = csvReader.readNext()) != null;) {
-                int fhrId = Integer.parseInt(line[fhrIdIndx]);
-                int rang = Integer.parseInt(line[rangIndx]);
-                TreeSet<Integer> rangSet = rangMap.get(fhrId);
+                int col1ID = Integer.parseInt(line[col1Index]);
+                int rang = Integer.parseInt(line[rangeIndex]);
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 10]
```diff
@@ -72,23 +71,23 @@ public String call (com.rivescript.RiveScript rs, String[] args) {
...
+                TreeSet<Integer> rangSet = rangMap.get(col1ID);
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

File: ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java [hunk 11]
```diff
@@ -25,7 +25,7 @@
 @Component
 public class SSBDiagramHandler implements Subroutine {
 	@Autowired
-	private FileUtil DemoMain;
+	private FileUtil fileUtil;
 	@Autowired
 	private ResponseBean responseBean;
     private Map<Integer, TreeSet<Integer>> rangMap = new HashMap<>();
```
Comment (Cortys) [inline ida-ws/src/main/java/upb/ida/provider/SSBDiagramHandler.java]: rename

**Comment by maqboolkhan** [comment]
Thanks again done :)

**Pull request merged**